### PR TITLE
Merge `master` to `time_series` 

### PR DESCRIPTION
**Pull request by Yard1**
## Related Issuse or bug

Info about Issue or bug

Partially completes #2374 

#### Describe the changes you've made

A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

File: pycaret/time_series/forecasting/functional.py [hunk 1]
```diff
@@ -4,20 +4,21 @@
...
 
 import numpy as np
 import pandas as pd
 
-from sktime.forecasting.base import ForecastingHorizon
-
 from pycaret.internal.utils import check_if_global_is_not_none
 from pycaret.time_series import TSForecastingExperiment
 
+if TYPE_CHECKING:
```
Comment (ngupta23) [inline pycaret/time_series/forecasting/functional.py:15]: Why is this needed?

Comment (Yard1) [reply pycaret/time_series/forecasting/functional.py:15]: This allows to avoid a circular dependency we would normally have. It will always be false during actual runtime, but true when handled by a 3rd party linter, allowing for type checking to work as intended.

https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

**Review by moezali1 — APPROVED** [review]

**Comment by ngupta23** [comment]
@Yard1 Do we need to check in the PNG and HTML files under the test folder? e.g. 

pycaret/tests/AUC.png & others.

**Comment by Yard1** [comment]
No, we don't. Thanks for pointing that out!

**Review by ngupta23 — APPROVED** [review]
